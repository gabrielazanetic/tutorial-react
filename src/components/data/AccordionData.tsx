import renderer from "react-test-renderer";
import Accordion from "../Accordion";

export const useAccordionData = () => {
	return renderer.create(<Accordion title="accordion">Hello</Accordion>);
};
