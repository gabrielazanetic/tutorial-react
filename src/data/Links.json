{
    "navigation": [
        {
            "id": "nav-1",
            "title": "Home page",
            "to": "/"
        },
        {
            "id": "nav-2",
            "title": "Tutorial",
            "to": "/tutorial"
        },
        {
            "id": "nav-3",
            "title": "Resources",
            "to": "/resources"
        }
    ],
    "tutorial-actions": [
        {
            "id": "tut-act-1",
            "title": "Copy URL",
            "to": "/tutorial"
        }
    ],
    "tutorial-jump": [
        {
            "id": "jump-to-step-1",
            "title": "Step 1 - Lorem ipsum",
            "to": "",
            "data-step": "1"
        },
        {
            "id": "jump-to-step-2",
            "title": "Step 2 - some more lorem ipsum",
            "to": "",
            "data-step": "2"
        },
        {
            "id": "jump-to-step-3",
            "title": "Step 3",
            "to": "",
            "data-step": "3"
        },
        {
            "id": "jump-to-step-4",
            "title": "Step 4",
            "to": "",
            "data-step": "4"
        },
        {
            "id": "jump-to-step-5",
            "title": "Step 5",
            "to": "",
            "data-step": "5"
        }
    ]
}