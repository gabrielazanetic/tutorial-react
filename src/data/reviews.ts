export const reviews = [
    {
        id: 1,
        date: new Date().setDate(new Date().getDate() - 1),
        title: "First Review",
        score: "****",
        review: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
    },
    {
        id: 2,
        date: new Date().setDate(new Date().getDate() - 2),
        title: "A Review",
        score: "***",
        review: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
    },
    {
        id: 3,
        date: new Date().setDate(new Date().getDate() - 3),
        title: "Last Review",
        score: "**",
        review: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
    },
    {
        id: 4,
        date: new Date().setDate(new Date().getDate() - 15),
        title: "Another Review",
        score: "*****",
        review: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
    },
    {
        id: 5,
        date: new Date().setDate(new Date().getDate() - 7),
        title: "Yet Another Review",
        score: "*",
        review: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
    },
    {
        id: 6,
        date: new Date().setDate(new Date().getDate() - 10),
        title: "A Review For The Tutorial",
        score: "**",
        review: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
    },
    {
        id: 7,
        date: new Date().setDate(new Date().getDate() - 11),
        title: "Reviews Abound",
        score: "**",
        review: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
    },
];