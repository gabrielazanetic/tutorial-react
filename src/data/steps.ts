export const steps = [
    {
        id: 1,
        title: "Figuring out the component hierarchy",
        screenshot: null,
        text: [
            "Before we jump into coding, the first thing to do is to figure out the component hierarchy from a provided mock or design. In our case, our mock will be this page you are currently reading from.",
            "We have a header with three different pieces of content, namely the logo, the main navigation and the user actions.",
            "Next, we have the main section of the page, which is split into two columns, one being the sidebar and the other being the steps to this tutorial."
        ]
    },
    {
        id: 2,
        title: "Starting the project",
        screenshot: null,
        text: [
            "The next thing to do is to check if nodejs is installed with the correct version. The version used for this project is 16.16.0.",
            "If the correct version is installed, then we can proceed to creating the project. There are several ways of doing this, and for the purpose of this project we will be using create-react-app and specify that we want to use TypeScript for developing the project, by using the keyword \"template\".",
            "Then you switch your current working directory to your project folder, run the startup script and start coding away."
        ],
        code: [
            "npx create-react-app app-name --template typescript",
            "cd app-name && npm start"
        ]
    },
    
    {
        id: 3,
        title: "Step 3",
        screenshot: null,
        text: [
            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
        ]
    },
    {
        id: 4,
        title: "Step 4",
        screenshot: null,
        text: [
            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
        ]
    },
    {
        id: 5,
        title: "Step 5",
        screenshot: null,
        text: [
            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
        ]
    }
];