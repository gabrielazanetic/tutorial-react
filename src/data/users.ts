export const users = [
    {
        id: 1,
        username: "admin",
        password: "testtest",
        fullname: "Gabriela Zanetic",
        birthday: new Date("1993-01-20"),
    }
];