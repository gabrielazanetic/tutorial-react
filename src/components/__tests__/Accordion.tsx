import { useAccordionData } from "../data/AccordionData";
import renderer from "react-test-renderer";

it("opens and closes accordion", () => {
	const accordion = useAccordionData();

	let tree: any = accordion.toJSON();
	expect(tree).toMatchSnapshot();

	console.log(tree.children);

	renderer.act(() => {
		tree.children[0].props.onClick();
	});

	tree = accordion.toJSON();
	expect(tree).toMatchSnapshot();

	renderer.act(() => {
		tree.children[0].props.onClick();
	});

	tree = accordion.toJSON();
	expect(tree).toMatchSnapshot();
});
