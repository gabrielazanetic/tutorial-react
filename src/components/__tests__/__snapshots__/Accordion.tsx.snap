// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`opens and closes accordion 1`] = `
<div
  className="accordion"
>
  <div
    className="accordion__header"
    onClick={[Function]}
  >
    <h3>
      accordion
    </h3>
  </div>
  <div
    className="accordion__body"
  >
    Hello
  </div>
</div>
`;

exports[`opens and closes accordion 2`] = `
<div
  className="accordion"
>
  <div
    className="accordion__header"
    onClick={[Function]}
  >
    <h3>
      accordion
    </h3>
  </div>
  <div
    className="accordion__body open"
  >
    Hello
  </div>
</div>
`;

exports[`opens and closes accordion 3`] = `
<div
  className="accordion"
>
  <div
    className="accordion__header"
    onClick={[Function]}
  >
    <h3>
      accordion
    </h3>
  </div>
  <div
    className="accordion__body"
  >
    Hello
  </div>
</div>
`;
